src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -174,7 +174,116 @@ static status_t del_routes_from_routing_table(private_kernel_super_ipsec_t* this
+									kernel_ipsec_policy_id_t* id,
+									ike_super_peer_info_t* super_peer_info,
+									child_sa_t * child_sa,
+									bool write_uptime) {
+	if(IKE_SUPER_PEER_INFO_TYPE_TUNNEL == super_peer_info->get_type(super_peer_info)) {
+		return;
+	}
+	ike_sa_t* const ike_sa = child_sa->get_ike_sa(child_sa);
+	ike_sa_id_t * const ike_sa_id = ike_sa->get_id(ike_sa);
+	struct oper_data oper_data;
+
+	uint16_t enc_alg = ENCR_UNDEFINED, int_alg = AUTH_UNDEFINED, size;
+	uint16_t esn = NO_EXT_SEQ_NUMBERS;
+	child_sa->get_proposal(child_sa)->get_algorithm(child_sa->get_proposal(child_sa), ENCRYPTION_ALGORITHM,
+								  &enc_alg, &size);
+	child_sa->get_proposal(child_sa)->get_algorithm(child_sa->get_proposal(child_sa), INTEGRITY_ALGORITHM,
+								  &int_alg, &size);
+	child_sa->get_proposal(child_sa)->get_algorithm(child_sa->get_proposal(child_sa), EXTENDED_SEQUENCE_NUMBERS,
+								  &esn, NULL);
+
+	oper_data.encryption_algorithm = enc_alg;
+	oper_data.integrity_algorithm = int_alg;
+	oper_data.ext_seq_num = esn;
+	oper_data.dh_group = ike_sa->get_ike_cfg(ike_sa)->get_dh_group(ike_sa->get_ike_cfg(ike_sa));
+	oper_data.anti_replay_window_size = child_sa->get_config(child_sa)->get_replay_window(child_sa->get_config(child_sa));
+
+	oper_data.version = ike_sa_id->get_ike_version(ike_sa_id);
+	oper_data.inbound_ipsec_spi = child_sa->get_spi(child_sa, 1);
+	oper_data.outbound_ipsec_spi = child_sa->get_spi(child_sa, 0);
+	oper_data.inbound_ike_spi = ike_sa_id->get_initiator_spi(ike_sa_id);
+	oper_data.outbound_ike_spi = ike_sa_id->get_responder_spi(ike_sa_id);
+	oper_data.nat_identified = child_sa->has_encap(child_sa);
+	oper_data.status = ike_sa->get_state(ike_sa);
+	oper_data.activeness_status = ike_sa->get_is_active(ike_sa);
+
+	char device_public_addr[MAX_IP_STR_LEN + 1];
+	char remote_public_addr[MAX_IP_STR_LEN + 1];
+	char local_network[MAX_IP_PREFIX_STR_LEN +1];
+	data->src->address_to_string(data->src, &device_public_addr[0]);
+	data->src->address_to_string(data->dst, &remote_public_addr[0]);
+	ts_to_prefix_string_for_logs(id->dst_ts, &local_network[0]);
+
+	oper_data.device_public_address = &device_public_addr[0];
+	oper_data.remote_public_address = &remote_public_addr[0];
+	oper_data.device_public_src_port = data->src->get_port(data->src);
+	oper_data.local_network = &local_network[0];
+
+	oper_data.account_id = super_peer_info->get_account_id(super_peer_info);
+	oper_data.vrf_name = super_peer_info->get_vrf_name(super_peer_info);
+	oper_data.device_id = super_peer_info->get_device_id(super_peer_info);
+
+	if(! oper_data.account_id || ! oper_data.vrf_name || !oper_data.device_id) {
+		LOG(DBG_KNL, SL_LOG_ERR, "super_peer_info has NULL values account-id: %s, vrf-name: %s, device-id: %s", oper_data.account_id, oper_data.vrf_name, oper_data.device_id);
+		goto out;
+	}
+
+	struct pm_terminator_conn* terminator_conn = NULL;
+	terminator_conn = lookup_terminator_conn(this, _get_sp_lookup_addr(data, id));
```

Comment:
Missing check if `lookup_terminator_conn()` returned NULL.

Reply:
Also note that the terminator will not be in the table in case the terminator is now being deleted and all of its tunnels are being deleted.